src/lib/converter/nodes/constructor.ts

```diff
@@ -42,7 +42,10 @@ export class ConstructorConverter extends ConverterNodeComponent<ts.ConstructorD
             if (!hasBody || !method.signatures) {
                 var name = 'new ' + parent.name;
                 var signature = createSignature(context, node, name, ReflectionKind.ConstructorSignature);
-                signature.type = new ReferenceType(parent.name, ReferenceType.SYMBOL_ID_RESOLVED, parent);
+                 // If no return type defined, then use the parent one.
+                if (!node.type) {
```

Comment:
What about checking `signature.type` instead of `node.type`? It seems that there is an `else` condition in https://github.com/TypeStrong/typedoc/blob/22ccb660ae6eb5b3e136dc277ecc1d62fb04d016/src/lib/converter/factories/signature.ts#L65 but the `.type` is being completely erased anyway?